android-app/app/src/main/java/de/dhbw/ka/se/fibo/CreateAccountActivity.java

```diff
@@ -55,9 +97,95 @@ private void initializeButton() {
     }
 
     private void registerButtonClicked() {
-        Intent i = new Intent(CreateAccountActivity.this,
+        i = new Intent(CreateAccountActivity.this,
                 MainActivity.class);
-        startActivity(i);
-        finish();
+        create_account_email = findViewById(R.id.create_account_email);
+        String email = create_account_email.getText().toString();
+        passwordField = findViewById(R.id.create_account_password_layer);
+        emailField = findViewById(R.id.create_account_email_layer);
+        create_account_password = findViewById(R.id.create_account_password);
+        String password = create_account_password.getText().toString();
+        createUser(email, password);
+
+    }
+
+    private void createUser(String email, String password) {
+        String url = "http://10.0.2.2:8000/users/register/";
+
+        RequestQueue queue = Volley.newRequestQueue(this);
```

Comment:
Please only initiate a new request queue if we are certain we'll fire a request. So, you could put it the if-guard checking whether the input is valid. This cleans up the code further.

Reply:
in what way valid ? check if the value is valid or if it is in the correct format ?

Reply:
you can put it after the if-branch (so, directly after the early return)